### ISSUE TITLE
Service Worker forces 401 status when using HTTP-Authentication

### ISSUE THREAD

**Issue by NCC1701M** [angular/angular#23605]
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[X] Bug report  
[ ] Performance issue
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
[ ] Other... Please describe:
</code></pre>

## Current behavior
When using the service worker with web apps which are using HTTP authentication (in this case Windows authentication) the ngsw.json?ngsw-cache-bust= runs into a 401 status.

## Expected behavior
The 401 status should not be happen.

## Minimal reproduction of the problem with instructions
Create a angular service worker project like described here [https://angular.io/guide/service-worker-getting-started](https://angular.io/guide/service-worker-getting-started) enable some kind of Authentication e.g. Windows authentication

## What is the motivation / use case for changing the behavior?
The motivation is that service workers should work with authentication.

# Possible solution
In the ngsw-worker.js file change the implementation of the newRequest method of the Adapter class to the following
````ts
newRequest(input, init) {
  init = init || {};
  if (!init.credentials) {
    init.credentials = 'same-origin';
  }
  return new Request(input, init);
}
````

## Environment

<pre><code>
Angular version: 5.2.10

Browser:
- [X] Chrome (desktop) version XX
- [ ] Chrome (Android) version XX
- [ ] Chrome (iOS) version XX
- [ ] Firefox version XX
- [ ] Safari (desktop) version XX
- [ ] Safari (iOS) version XX
- [ ] IE version XX
- [ ] Edge version XX
 
For Tooling issues:
- Node version: 8.9.4
- Platform:  Windows

**Comment by NCC1701M** [comment]
@gkalpak Yeah, well then let the developers decide if they want HTTP-Authentication or not and implement a flag which they can set to enable it.

**Comment by gkalpak** [comment]
@NCC1701M, I'm with you on enabling it behind a flag (unless @alxhub convinces me otherwise).
Just adding a reference to the previous issue, which has related discussion and people might not be aware of.

**Comment by inthegarage** [comment]
@gkalpak I can confirm this works for my situation, as mentioned on the linked ticket. Thank you.

**Comment by kaihenzler** [comment]
@NCC1701M thanks for the workaround. I ended up patching `ngsw-worker.js` like you proposed. This is kind of a blocker for my project right now. We migrated from a custom webpack build to the angular-cli. In our custom buildsystem we integrated the https://www.npmjs.com/package/workbox-webpack-plugin which in fact was able to deal with HTTP auth. 

I thought about creating a PR but it will most likely be refused due to #23425 being closed already.

Are there any plans to integrate a similar solution as described in this issue or will HTTP auth never be supported by the angular-cli?

**Comment by alxhub** [comment]
@gkalpak I've been out of SW development for a while now, so my knowledge here is a bit rusty. But my impression was that `credentials = 'same-origin'` is the default behavior if unspecified. So is this still an issue?

**Comment by j2L4e** [comment]
It seems this is no longer an issue. I jsut tried it and the worker itself ran out of the box.
For the manifest to load you'll need to change the manifest link in `index.html` though:

```html
<link rel="manifest" href="manifest.webmanifest" crossorigin="use-credentials">
```
See https://developer.mozilla.org/en-US/docs/Web/Manifest

**Comment by j2L4e** [comment]
Well, scratch that. As soon as your initial BasicAuth expires any requests to the server will still fail with 401, but the prompt doesn't show up. Not sure if that's a problem with the package or a browser limitation, though.

**Comment by gkalpak** [comment]
Given that the default value for [Request#credentials][1] is indeed `same-origin`, I don't think there is anything to be fixed on the Angular side. As reported above, things work as expected in Firefox and from the two bugs reported for Chrome [623464][2] is closed as `fixed` and [1055253][3] is still pending (but will hopefully be fixed soon).

If you think there is something that could be fixed/improved in `@angular/service-worker`, please open a new issue.

[1]: https://developer.mozilla.org/en-US/docs/Web/API/Request/credentials
[2]: https://bugs.chromium.org/p/chromium/issues/detail?id=623464
[3]: https://bugs.chromium.org/p/chromium/issues/detail?id=1055253
